### PR TITLE
builder/tencentcloud: showing tencentcloud image id after copy to desination region

### PR DESCRIPTION
**Pull request by likexian**
## Tests passed

```
=== RUN   TestTencentCloudAccessConfig_Prepare
--- PASS: TestTencentCloudAccessConfig_Prepare (0.00s)
=== RUN   TestTencentCloudImageConfig_Prepare
--- PASS: TestTencentCloudImageConfig_Prepare (0.00s)
=== RUN   TestTencentCloudRunConfig_Prepare
--- PASS: TestTencentCloudRunConfig_Prepare (0.00s)
=== RUN   TestTencentCloudRunConfigPrepare_UserData
--- PASS: TestTencentCloudRunConfigPrepare_UserData (0.00s)
=== RUN   TestTencentCloudRunConfigPrepare_UserDataFile
--- PASS: TestTencentCloudRunConfigPrepare_UserDataFile (0.00s)
=== RUN   TestTencentCloudRunConfigPrepare_TemporaryKeyPairName
--- PASS: TestTencentCloudRunConfigPrepare_TemporaryKeyPairName (0.00s)
=== RUN   TestTencentCloudRunConfigPrepare_SSHPrivateIp
--- PASS: TestTencentCloudRunConfigPrepare_SSHPrivateIp (0.00s)
PASS
ok      github.com/hashicorp/packer/builder/tencentcloud/cvm    0.010s
```

## Packer build passed

```
tencentcloud-cvm: output will be in this color.

==> tencentcloud-cvm: Trying to check image name: copy_regions...
    tencentcloud-cvm: Image name: useable
==> tencentcloud-cvm: Trying to check source image: img-9qabwvbn...
    tencentcloud-cvm: Image found: CentOS 7.6 64bit
==> tencentcloud-cvm: Trying to create a new keypair: packer_5e4b5c15...
    tencentcloud-cvm: Keypair created: skey-r5hkre0v
==> tencentcloud-cvm: Trying to create a new vpc...
    tencentcloud-cvm: Vpc created: vpc-aq3zfi27
==> tencentcloud-cvm: Trying to create a new subnet...
    tencentcloud-cvm: Subnet created: subnet-fzxvzx5c
==> tencentcloud-cvm: Trying to create a new securitygroup...
    tencentcloud-cvm: Securitygroup created: sg-3o5x2sxb
==> tencentcloud-cvm: Trying to create securitygroup polices...
    tencentcloud-cvm: Securitygroup polices created
==> tencentcloud-cvm: Trying to create a new instance...
    tencentcloud-cvm: Waiting for instance ready
    tencentcloud-cvm: Instance created: ins-1l00y06a
==> tencentcloud-cvm: Using ssh communicator to connect: 134.175.144.77
==> tencentcloud-cvm: Waiting for SSH to become available...
==> tencentcloud-cvm: Connected to SSH!
==> tencentcloud-cvm: Provisioning with shell script: /tmp/packer-shell338308769
    tencentcloud-cvm: total 40K
    tencentcloud-cvm: -rw-------  1 root root    2 Feb 18 11:38 .bash_history
    tencentcloud-cvm: -rw-r--r--. 1 root root   18 Dec 29  2013 .bash_logout
    tencentcloud-cvm: -rw-r--r--. 1 root root  176 Dec 29  2013 .bash_profile
    tencentcloud-cvm: -rw-r--r--. 1 root root  176 Dec 29  2013 .bashrc
    tencentcloud-cvm: drwxr-xr-x  3 root root 4.0K Mar  7  2019 .cache
    tencentcloud-cvm: drwxr-xr-x  3 root root 4.0K Mar  7  2019 .config
    tencentcloud-cvm: -rw-r--r--. 1 root root  100 Dec 29  2013 .cshrc
    tencentcloud-cvm: -rw-------  1 root root   42 Nov  5 23:24 .lesshst
    tencentcloud-cvm: drwx------  2 root root 4.0K Nov  5 23:14 .ssh
    tencentcloud-cvm: -rw-r--r--. 1 root root  129 Dec 29  2013 .tcshrc
    tencentcloud-cvm: -rw-------  1 root root    0 Feb 18 11:38 .viminfo
==> tencentcloud-cvm: Trying to detach keypair: skey-r5hkre0v...
    tencentcloud-cvm: Keypair detached
==> tencentcloud-cvm: Trying to create a new image: copy_regions...
    tencentcloud-cvm: Waiting for image ready
    tencentcloud-cvm: Image created: img-qhjudzm0
==> tencentcloud-cvm: Trying to copy image to: ap-shanghai,ap-beijing,ap-chengdu...
    tencentcloud-cvm: Waiting for image ready
    tencentcloud-cvm: Copy image from ap-guangzhou(img-qhjudzm0) to ap-shanghai(img-om856nq1)
    tencentcloud-cvm: Copy image from ap-guangzhou(img-qhjudzm0) to ap-beijing(img-k82zda2d)
    tencentcloud-cvm: Copy image from ap-guangzhou(img-qhjudzm0) to ap-chengdu(img-mz85q4tk)
    tencentcloud-cvm: Image copied
==> tencentcloud-cvm: Cleaning up instance...
==> tencentcloud-cvm: Cleaning up securitygroup...
==> tencentcloud-cvm: Cleaning up subnet...
==> tencentcloud-cvm: Cleaning up vpc...
==> tencentcloud-cvm: Cleaning up keypair...
Build 'tencentcloud-cvm' finished.

==> Builds finished. The artifacts of successful builds are:
--> tencentcloud-cvm: Tencentcloud images(ap-beijing: img-k82zda2d
ap-chengdu: img-mz85q4tk
ap-guangzhou: img-qhjudzm0
ap-shanghai: img-om856nq1) were created.

```

### CODE REVIEW

File: builder/tencentcloud/cvm/common.go [hunk 1]
```diff
@@ -10,16 +10,18 @@ import (
...
+	vpc "github.com/tencentcloud/tencentcloud-sdk-go/tencentcloud/vpc/v20170312"
 )
 
 // DefaultWaitForInterval is sleep interval when wait statue
 const DefaultWaitForInterval = 5
 
 // WaitForInstance wait for instance reaches statue
-func WaitForInstance(client *cvm.Client, instanceId string, status string, timeout int) error {
-	ctx := context.TODO()
+func WaitForInstance(ctx context.Context, client *cvm.Client, instanceId string, status string, timeout int) error {
```
Comment (azr) [inline builder/tencentcloud/cvm/common.go]: Nice !

File: builder/tencentcloud/cvm/common.go [hunk 2]
```diff
@@ -50,54 +52,86 @@ func WaitForInstance(client *cvm.Client, instanceId string, status string, timeo
...
-	req := cvm.NewDescribeImagesRequest()
-	FILTER_IMAGE_NAME := "image-name"
-	req.Filters = []*cvm.Filter{
-		{
-			Name:   &FILTER_IMAGE_NAME,
-			Values: []*string{&imageName},
-		},
-	}
-
+func WaitForImageReady(ctx context.Context, client *cvm.Client, imageName string, status string, timeout int) error {
```
Comment (azr) [inline builder/tencentcloud/cvm/common.go]: nice !

File: builder/tencentcloud/cvm/common.go [hunk 3]
```diff
@@ -50,54 +52,86 @@ func WaitForInstance(client *cvm.Client, instanceId string, status string, timeo
...
 	for {
-		var resp *cvm.DescribeImagesResponse
-		err := Retry(ctx, func(ctx context.Context) error {
-			var e error
-			resp, e = client.DescribeImages(req)
-			return e
-		})
+		image, err := GetImageByName(ctx, client, imageName)
```
Comment (azr) [inline builder/tencentcloud/cvm/common.go]: Here for retries you could use something like:

https://github.com/hashicorp/packer/blob/39c25b2c661a2565f2506476ea292ac81507dd36/builder/amazon/common/helper_funcs.go#L54-L65

File: builder/tencentcloud/cvm/common.go [hunk 4]
```diff
@@ -50,54 +52,86 @@ func WaitForInstance(client *cvm.Client, instanceId string, status string, timeo
...
 }
 
 // CheckResourceIdFormat check resource id format
 func CheckResourceIdFormat(resource string, id string) bool {
 	regex := regexp.MustCompile(fmt.Sprintf("%s-[0-9a-z]{8}$", resource))
-	if !regex.MatchString(id) {
-		return false
-	}
-	return true
+	return regex.MatchString(id)
```
Comment (azr) [inline builder/tencentcloud/cvm/common.go]: 👍

File: builder/tencentcloud/cvm/common.go (hunk 3 above)
Comment (likexian) [reply builder/tencentcloud/cvm/common.go]: Sure! I have a helper function defined here:
https://github.com/hashicorp/packer/blob/39c25b2c661a2565f2506476ea292ac81507dd36/builder/tencentcloud/cvm/common.go#L116

Comment (azr) [reply builder/tencentcloud/cvm/common.go]: Ah I didn't open it ! 👍  LGTM then :D

Comment (likexian) [reply builder/tencentcloud/cvm/common.go]: Thanks anyway :)

**Pull request merged**